vscode-trace-extension/src/utils/trace-extension-event-manager.ts

```diff
@@ -0,0 +1,69 @@
+/***************************************************************************************
+ * Copyright (c) 2023 BlackBerry Limited and others.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ ***************************************************************************************/
+import * as vscode from 'vscode';
+
+export interface SignalCallbackInterface {
+    signal: string,
+    callback: (data: unknown) => void
+}
+
+/**
+ *  Manages the events for the trace extension
+ */
+export class TraceExtensionEventManager {
+    private eventEmitterMap: Map<string, vscode.EventEmitter<unknown>>;
+    constructor() {
+        this.eventEmitterMap = new Map<string, vscode.EventEmitter<unknown>>();
+    }
+
+    /**
+     * Get the event emitter registered based on Id
+     *
+     * @param id event emitter Id
+     * @returns event emitter for the given id if exists, otherwise undefined
+     */
+    getEventEmitter(id: string): vscode.EventEmitter<unknown> | undefined {
+        return this.eventEmitterMap.get(id);
+    }
+
+    /**
+     * Register an event emitter with the provided Id
+     *
+     * @param id event emitter Id
+     */
+    registerEventEmitter(id: string): void {
+        this.eventEmitterMap.set(id, new vscode.EventEmitter<unknown>());
+    }
```

Comment:
Should this return a `Disposable` allowing the creator of the `EventEmitter` to dispose of it and remove it from the `Map`?